lib/rpc-client.js

```diff
@@ -0,0 +1,60 @@
+import {
+  FilecoinRpcError,
+  INVALID_RPC_RESPONSE,
+  NO_RESULT_IN_RESPONSE,
+  RPC_ERROR,
+} from './errors.js'
+
+/**
+ * @param {string} method
+ * @param {unknown[]} params
+ * @param {string} rpcUrl
+ * @param {string} rpcAuth
+ * @param {object} [options]
+ * @param {globalThis.fetch} [options.fetch]
+ */
+export async function rpc(method, params, rpcUrl, rpcAuth, { fetch = globalThis.fetch } = {}) {
+  const req = new Request(rpcUrl, {
+    method: 'POST',
+    headers: {
+      'content-type': 'application/json',
+      accepts: 'application/json',
+      authorization: `Bearer ${rpcAuth}`,
+    },
+    body: JSON.stringify({
+      jsonrpc: '2.0',
+      id: 1,
+      method,
+      params,
+    }),
+  })
+  const res = await fetch(req, {
+    signal: AbortSignal.timeout(60_000),
```

Comment:
Please make the timeout configurable.

I think the best solution is to change this function to accept any `signal` and let the caller create a signal triggered by a timeout.

```js
export async function rpc(method, params, { rpcUrl, rpcAuth, signal, fetch = globalThis.fetch } = {})
  // ...
  const res = await fetch(req, { signal })
}

// usage
const signal = AbortSignal.timeout(60_000)
await rpc(method, params, { signal, /* ... */ })
```

This is a pattern often used in modern JavaScript.